app/src/features/apiClient/screens/apiClient/APIClient.tsx

```diff
@@ -62,7 +99,12 @@ export const APIClient: React.FC<Props> = () => {
           onNewClick={onNewClick}
           onImportClick={onImportClick}
         />
-        <APIClientView apiEntry={selectedEntry} notifyApiRequestFinished={addToHistory} />
+        <APIClientView
+          key={requestId}
+          apiEntry={selectedEntry ?? selectedEntryDetails?.data}
+          apiEntryDetails={selectedEntryDetails}
```

Comment:
Two similar props are beings passed. Can we have a single prop which does the work for both these props?

Reply:
Its intentional to avoid the refactor required for history, since currently we only store the entry in localStorage instead we need to store the whole record with id.

Reply:
Can we rename the props by the time it gets refactored? so that its clear which prop is to be used from history and which one for saved requests